scalableminds/webknossos
# Make layer alpha configurable and generalize Rgb/Grayscale layer handling

- Got rid of the `abstract_plane_material_factory` which was a legacy OOP artifact
- Added an `alpha` property to the data layer configuration, which can be used to fade the different layers in and out
- Generalized the handling of rgb and grayscale layers, so that rgb layers can now have any name (previously they needed to be named `color`) and more importantly a dataset can now consist of a mix of rgb and grayscale layers

Ideally, we should also avoid loading and prefetching data for layers whose alpha value is 0 (same as for the segmentation). To do that, I think it would be very beneficial to soften the specialized handling of the segmentation layer somewhat, which would make it easier to use the existing code for all layers. I'll do that in another PR (Issue: #3671) :)

### URL of deployed dev instance (used for testing):
- https://layeropacity.webknossos.xyz

### Steps to test:
- Open a dataset with multiple layers and use the alpha slider in the dataset settings.
- Create a dataset with an RGB and grayscale layer. Open it - all layers should be visible.
- [x] The screenshot tests are green (I'll test that)

### Issues:
- fixes #3626 

------
